sdk/include/reactos/shellutils.h

```diff
@@ -227,25 +227,23 @@ class CComDebugObject : public Base
 #endif
 
 template<class T>
-void ReleaseCComPtrExpectZero(CComPtr<T>& cptr, BOOL forceRelease = FALSE)
+ULONG ReleaseCComPtrExpectZeroHelper(const char *file, UINT line, CComPtr<T>& cptr, BOOL forceRelease = FALSE)
 {
+    ULONG r = 0;
     if (cptr.p != NULL)
     {
         T *raw = cptr.Detach();
-        int nrc = raw->Release();
+        int nrc = r = raw->Release();
         if (nrc > 0)
+            Win32DbgPrint(file, line, "WARNING: Unexpected RefCount > 0 (%d)\n", nrc);
+        while (nrc > 0 && forceRelease)
         {
-            DbgPrint("WARNING: Unexpected RefCount > 0 (%d)!\n", nrc);
-            if (forceRelease)
-            {
-                while (nrc > 0)
-                {
-                    nrc = raw->Release();
-                }
-            }
+            nrc = raw->Release();
         }
     }
+    return r;
 }
+#define ReleaseCComPtrExpectZero(...) ReleaseCComPtrExpectZeroHelper(__FILE__, __LINE__, __VA_ARGS__)
```

Comment:
```suggestion
#define ReleaseCComPtrExpectZero(...) ReleaseCComPtrExpectZeroHelper(__FILE__, __LINE__, ##__VA_ARGS__)
```
The reason for the extra two `##`: https://stackoverflow.com/questions/52891546/what-does-va-args-mean

Reply:
That post says it's a GCC extension, what about msvc?

Not that it matters, you can't call this with zero arguments, what would you be releasing?

Reply:
I would just say, check what we do in other parts of _our_ (not 3rd-party) codebase.
Obviously it works with MSVC since we use it in other places.

Reply:
Good to know. But in this case, it does not matter since you have to pass at least one parameter.

IMHO vararg is pointless here because the option to call Release until it returns 0 breaks COM rules.

If it was up to me, ReleaseCComPtrExpectZero would not exist at all. If you pass a valid object to someone, they are allowed to AddRef and hold on to the interface longer than "you".

Reply:
To express (and have the compiler understand) that the macro expects one parameter or more, here's the updated suggestion:
```suggestion
#define ReleaseCComPtrExpectZero(cptr, ...) ReleaseCComPtrExpectZeroHelper(__FILE__, __LINE__, cptr, ##__VA_ARGS__)
```
that works in both GCC and MSVC (and our other supported compilers).